File: java/word-count/src/main/java/WordCount.java
import java.util.Map;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public final class WordCount {
    public Map<String, Integer> Phrase(String line) {
        return Stream.of(line.toLowerCase().split("\\W+"))
                     .collect(Collectors.toMap(
                        x -> x,
                        x -> 1,
                        Integer::sum));
    }
}

File: v1/kotlin/word-count/src/main/kotlin/WordCount.kt
object WordCount {
    fun phrase(input: String): Map<String, Int> {
        return input.toLowerCase()
                .filter { Character.isLetterOrDigit(it) || it == ' ' || it == '\'' }
                .split(Regex(" +"))
                .groupBy { it }
                .map { Pair(it.key, it.value.size )}
                .toMap()
    }
}
